compiler/resolution/implementForallIntents.cpp

```diff
@@ -1062,6 +1062,152 @@ static const char* astrArg(int ix, const char* add1) {
+    ... yield (555, addr-of(x1_shadowVarReduc)); ...
+    x1_reduceParent.accumulate(x1_shadowVarReduc);
+  }
+
+Particular Concern
+------------------
+
+If the above yield is actually within a forall:
+
+  forall ... {
+    ... yield (555, addr-of(x1_shadowVarReduc)); ...
+  }
+
+then we need to create a reduce-intented shadow variable for this
+forall, corresponding to x1_shadowVarReduc, and yield it instead:
+
+  forall ... with (x1_reduceParent reduce x1_shadowVarReduc) {
+    // here, "x1_shadowVarReduc" is the shadow variable:
+    ... yield (555, addr-of(x1_shadowVarReduc)); ...
+  }
+
+Look Out For...
+---------------
+
+Need to handle these cases:
+
+* The yield is in >1 nested foralls --> create a new shadow var for each:
+
+  iter myIter(param tag == standlone, x1_reduceParent:SumReduceScanOp) {
+    var x1_shadowVarReduc = x1_reduceParent.identity;
+
+    forall ... with (x1_reduceParent reduce x1_shadowVarReduc) {
+      // shadow var: x1_shadowVarReduc'
+
+      forall ... with (x1_reduceParent reduce x1_shadowVarReduc') {
+        // shadow var: x1_shadowVarReduc''
+
+        ... yield (555, addr-of(x1_shadowVarReduc'')); ...
+      }
+    }
+
+    x1_reduceParent.accumulate(x1_shadowVarReduc);
+  }
+
+* Multiple yields within the same forall(s) --> use the same shadow var:
+
+    forall ... with (x1_reduceParent reduce x1_shadowVarReduc) {
+      // shadow var: x1_shadowVarReduc'
+
+      ... yield (555, addr-of(x1_shadowVarReduc'')); ...
+      ... yield (666, addr-of(x1_shadowVarReduc'')); ...
+    }
+
+*/
+static Symbol* createShadowVarIfNeeded(ShadowVarSymbol *shadowvar,
+                                       Symbol* parentOp, Symbol* svar,
+                                       CallExpr* yieldCall)
+{
+  ForallStmt* efs = enclosingForallStmt(yieldCall);
+  if (!efs)
+    // the original variable is just fine
+    return svar;
+
+  // In the "old" world, we call this only for reductions.
+  ForallIntentTag intent = shadowvar ? shadowvar->intent : TFI_REDUCE;
+
+  if (ForallStmt* efs2 = enclosingForallStmt(efs)) {
+    // TODO arrange for a chain of shadow variables.
+    // This should be straightforward. Presently not implemented.
+
+    switch (intent) {
+      case TFI_CONST_IN:
+      case TFI_REF:
+      case TFI_CONST_REF:
+        // For these intents, OK to go without a new shadow variable.
+        return svar;
+
+      case TFI_IN:
+      case TFI_REDUCE:
+        // May result in data races or incorrect behavior
+        // if we don't handle multiple enclosing foralls.
+        USR_FATAL_CONT(yieldCall, "a parallel iterator with a yield nested in two or more enclosing forall statements is not currently implemented in the presence of an 'in' or 'reduce' intent.");
+        USR_PRINT(efs, "the immediately enclosing forall statement is here");
+        USR_PRINT(efs2, "the next enclosing forall statement is here");
+        // If we continue, we may get a const violation on x*_shadowVarReduc.
+        USR_STOP();
```

Comment:
Shouldn't this error be triggered in the new test you added? Or does it only fire for multiple nested foralls like this?
```
iter ( ... )
  forall ...
    forall ...
      yield ...
```

Reply:
This should not be triggered in my test. As you say, the error requires at least 2 nested foralls around the yield statement.